iml-manager-cli/src/api.rs

```diff
@@ -44,8 +45,14 @@ pub async fn api_cli(command: ApiCommand) -> Result<(), ImlManagerCliError> {
         ApiMethod::Put => client.put(uri),
     };
 
-    let body: Option<serde_json::Value> =
-        command.body.map(|s| serde_json::from_str(&s)).transpose()?;
+    let body: Option<serde_json::Value> = if command.body == Some("-".to_string()) {
+        let mut buf: Vec<u8> = Vec::new();
+        stdin().read_to_end(&mut buf).await?;
+        let s = String::from_utf8_lossy(&buf);
+        Some(serde_json::from_str(&s)?)
```

Comment:
You could probably convert the result to an option here.
```
serde_json::from_str(&s).ok()
```

Reply:
That would lose whatever the error was though